### PR TITLE
Automate call to Lighthouse polling job

### PR DESCRIPTION
**Pull request by mchae-nava**
## Summary

- *This work is behind a feature toggle (flipper): YES*

Relatively simple change. One submitted, we will test in staging behind a Flipper to ensure that the jobs still runs properly

Context: 

With the migration to the Lighthouse synchronous submit endpoint, Lighthouse returns after a claim is established and returns a claim id from VBMS representing the established claim.
However, this does not ensure that the Form 526 PDF that was generated in the process got submitted along with the claim being established.
If we were to utilize the Lighthouse synchronous submit endpoint, we would not know if the generated Form 526 PDF made it into the eFolder for the Veteran in VBMS.

This PR includes:
- Automating polling implementation
- Unit Tests
- Feature Flag
- Addendum to job to exit if created_at date is more than 2 days ago

## Related issue(s)

- *Link to ticket created in va.gov-team repo OR screenshot of Jira ticket if your team uses Jira*
https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/85624

## Testing done

- [x] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: app/models/form526_submission.rb [hunk 1]
```diff
@@ -563,6 +566,12 @@ def submit_flashes
     BGS::FlashUpdater.perform_async(id) if user && Flipper.enabled?(:disability_compensation_flashes, user)
   end
 
+  def poll_form526_pdf
+    # In order to track the status of the 526 PDF upload via Lighthouse,
+    # call poll_form526_pdf, provided we received a valid claim_id from Lighthouse
+    Lighthouse::PollForm526Pdf.perform(id) if submitted_claim_id
```
Comment (sethdarragile6) [inline app/models/form526_submission.rb:572]: Shouldn't this be `.perform_async(id)` instead?

File: app/sidekiq/lighthouse/poll_form526_pdf.rb [hunk 2]
```diff
@@ -76,6 +76,20 @@ def perform(submission_id) # rubocop:disable Metrics/MethodLength
       Sentry.set_tags(source: '526EZ-all-claims')
 
       with_tracking('Form526 Submission', submission.saved_claim_id, submission.id, submission.bdd?) do
+        # Check the submission.created_at date, if it's more than 2 days old and the PDF is missing
```
Comment (sethdarragile6) [inline app/sidekiq/lighthouse/poll_form526_pdf.rb:79]: I think we want this check further down in the method. We don't want to set the job status to `pdf_not_found` _before_ we even check whether or not the PDF exists. I was thinking more in the else condition, where normally we would "keep on retrying" but in _this_ case- don't. It's not there and it's older than 48 hours.

**Review by sethdarragile6 — CHANGES_REQUESTED** [review]
A couple questions, suggested tweaks. Otherwise looks good!

File: app/sidekiq/lighthouse/poll_form526_pdf.rb (hunk 2 above)
Comment (mchae-nava) [reply app/sidekiq/lighthouse/poll_form526_pdf.rb:79]: Oh I see, I was understanding the ask as "exit before doing anything meaningful if we already know we're past the limit"

Got it, will make the adjustment

File: app/models/form526_submission.rb (hunk 1 above)
Comment (mchae-nava) [reply app/models/form526_submission.rb:572]: Whoops, a relic from my local testing. Will fix!

**Review by sethdarragile6 — APPROVED** [review]

File: app/sidekiq/lighthouse/poll_form526_pdf.rb [hunk 3]
```diff
@@ -96,6 +96,18 @@ def perform(submission_id) # rubocop:disable Metrics/MethodLength
...
         else
+          # Check the submission.created_at date, if it's more than 2 days old
+          # update the job status to pdf_not_found immediately and exit the job
+          unless submission.created_at.between?(Date.current - 2, Date.current)
+            form_job_status.update(
+              status: Form526JobStatus::STATUS[:pdf_not_found],
+              bgjob_errors: bgjob_errors.merge(new_error)
+            )
+            ::Rails.logger.warn(
+              'Submission creation date is over 2 days old. Exiting...'
```
Comment (rmtolmach) [inline app/sidekiq/lighthouse/poll_form526_pdf.rb:107]: Can you make this message more specific so when you see it amongst all the other warnings you know it's from form 526?

Comment (mchae-nava) [reply app/sidekiq/lighthouse/poll_form526_pdf.rb]: Absolutely!

**Review by sethdarragile6 — APPROVED** [review]

**Review by aurora-a-k-a-lightning — APPROVED** [review]

File: app/sidekiq/lighthouse/poll_form526_pdf.rb (hunk 3 above)
Comment (rmtolmach) [reply app/sidekiq/lighthouse/poll_form526_pdf.rb]: Thanks! Updates look good!

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**